docs/learn/learn-parachains-protocol.md

```diff
@@ -156,22 +295,233 @@ para-blocks are shown as yellow squares that become grey in color.
 
 ![parachain-approval-process](../assets/parachain-approval-process.png)
 
+### Assignments & Secondary Checks
+
+Having a bad parablock on the relay chain is not catastrophic as long as the block is not approved
+and finalized by the finality gadget [GRANDPA](./learn-consensus.md/#finality-gadget-grandpa). If
+the block is not finalized, the fork on the chain containing that block can be ignored in favor of
+another fork containing good blocks. Dealing with a bad parablock includes the following stages:
+
+- Detection: the bad block must be detected by honest validators.
+- Escalation: the honest validators must send that block for checks to all validators. A
+  [dispute](#disputes) starts.
+- Consequences: the chain is reverted and all malicious validators are slashed.
+
+The result of the dispute must be transplantable to all other forks so that malicious validators are
+slashed in all possible histories and so that honest validators will ignore any forks containing
+that parablock.
```

Comment:
Major red flags reading these paragraphs. What does it mean to be on the Relay Chain (also, be consistent with style) if not finalized? What is the difference between being ignored in favor of another fork vs. being reverted? Why would a block's validation influence sending that block to others? (I.e., the block should already be available, why would they need to send it? They can just send a dispute and others can request it to get involved.)